packages/cli/src/commands/composite/compile.ts

```diff
@@ -20,22 +23,43 @@ export default class CompositeCompile extends Command<Flags> {
       .map((token) => {
         return token.input
       })
-
-    if (allArgs.length < 2) {
-      this.spinner.fail('Missing composite path and at output path')
-      return
-    }
     try {
-      const [compositePath, ...outputPaths] = allArgs
-      const composite = await readEncodedComposite(this.ceramic, compositePath)
+      let composite: Composite | undefined = undefined
+      let outputPaths: Array<string> = []
+      if (this.stdin !== undefined && allArgs.length >= 1) {
+        const definition = JSON.parse(this.stdin) as EncodedCompositeDefinition
+        composite = await Composite.fromJSON({ ceramic: this.ceramic, definition })
+        outputPaths = allArgs
+      } else if (this.stdin === undefined && allArgs.length >= 2) {
+        composite = await readEncodedComposite(this.ceramic, allArgs[0])
+        outputPaths = allArgs.splice(1)
+      } else if (this.stdin !== undefined && allArgs.length < 1) {
+        this.spinner.fail(
+          'When the composite is passed as JSON in stdin, at least one output path needs to be given as param'
+        )
+        return
+      } else {
+        this.spinner.fail('Missing composite path and at output path')
+        return
+      }
       const runtimeDefinition = composite.toRuntime()
       outputPaths.map(async (outputPath) => {
         await writeRuntimeDefinition(runtimeDefinition, outputPath)
       })
-      this.spinner.succeed('Successfully saved compiled composite into given path(s)')
+      this.spinner.succeed('Compiling the composite... Done!')
+      if (!this.flags['disable-stdin']) {
```

Comment:
Seems OCLIF can automatically handle reversing boolean flags with the `--no-` prefix (https://oclif.io/docs/flags), maybe the flag could be set to simply `stdin` with no default value, so when explicitly setting `--stdin` the logic should only be applied based on the input, `--no-stdin` should explicitly ignore the input and apply the argument-based logic, and not providing the flag should infer the logic based on the presence of an input or not?

Reply:
Actually, this code is a leftover and not used right now.

At some point in this PR I had the `--disable-stdin` flag implemented and just used in tests, because for some reason the `stdin` streams were not receiving `end` signals in tests (and only in tests), which made them fail with timeouts.

But then I realised that to make the tests work, I can just add a timeout in `readPipe()` function above and then I don't need the option to disable stdin and so I removed this flag (but left this piece of code by omission). This however, means that the the `readPipe()` function also has a timeout implemented, which makes it different from something like https://github.com/sindresorhus/get-stdin

So, to summarise, I see two options:
a. I can implement the `stdin` flag like you suggested in your comment and replace my `readPipe()` function with https://github.com/sindresorhus/get-stdin 
b. I can remove the leftover code that you commented on, leave my custom `readPipe()` function and there will be no option to disable stdin handling

Which one do you think will work better?

I think I prefer b., because with a. we'd have to set the `--no-stdin` flag in tests and then we wouldn't be testing the whole code path. But I don't feel super strongly about it.

Reply:
Putting the same answer here that I gave in the "Files Changed" tab (not sure why it doesn't appear here):

Actually, this code is a leftover and not used right now.

At some point in this PR I had the `--disable-stdin` flag implemented and just used in tests, because for some reason the stdin streams were not receiving end signals in tests (and only in tests), which made them fail with timeouts.

But then I realised that to make the tests work, I can just add a timeout in `readPipe()` function above and then I don't need the option to disable stdin and so I removed this flag (but left this piece of code by omission). This however, means that the the `readPipe()` function also has a timeout implemented, which makes it different from something like https://github.com/sindresorhus/get-stdin

So, to summarise, I see two options:
a. I can implement the stdin flag like you suggested in your comment and replace my `readPipe()` function with https://github.com/sindresorhus/get-stdin
b. I can remove the leftover code that you commented on, leave my custom `readPipe()` function and there will be no option to disable stdin handling

Which one do you think will work better?

I think I prefer b., because with a. we'd have to set the `--no-stdin` flag in tests and then we wouldn't be testing the whole code path. But I don't feel super strongly about it.

Reply:
So if I understand correctly, using `get-stdin` and `execa` together in tests doesn't work well, the process hangs?
Overall I think it could be useful to be able to force using or disable stdin via a flag, but no need to take care of it in this PR, so whatever you think is simpler to make the tests work as expected.

Reply:
> So if I understand correctly, using get-stdin and execa together in tests doesn't work well, the process hangs?

To be fair, I haven't checked this specifically with `get-stdin`, but I'm pretty sure it will hang after having a look at the code in that package.

I'll check it today and if it hangs, I'm gonna stick to my custom function for now and not implement the flag in this PR. If that's the case, I'll add a ticket to implement the flag and do this in the next cycle :).

Reply:
Ok, @PaulLeCam , I checked that the tests hang when replacing the custom `readPipe()` function with `get-stdin`.

You can have a look at this draft PR and the jobs there to confirm: https://github.com/ceramicstudio/js-glaze/pull/149

I propose to merge this PR as is then and I'll implement the `--stdin` flag next week. Does that work :) ?